pkg/cortex/cortex.go

```diff
@@ -238,7 +241,20 @@ func (t *Cortex) initModule(cfg *Config, m moduleName) error {
 
 // Run starts Cortex running, and blocks until a signal is received.
 func (t *Cortex) Run() error {
-	return t.server.Run()
+	var err error
+	done := make(chan struct{})
```

Comment:
We are not waiting for this anywhere (I will create a follow-up PR to your branch, just a note to myself)